src/engine/LoadURL.cpp

```diff
@@ -0,0 +1,147 @@
+// Copyright 2025 The QLever Authors, in particular:
+//
+// 2025 Julian Mundhahs <mundhahj@tf.uni-freiburg.de>, UFR
+//
+// UFR = University of Freiburg, Chair of Algorithms and Data Structures
+
+#include "engine/LoadURL.h"
+
+#include "global/RuntimeParameters.h"
+#include "util/http/HttpUtils.h"
+
+// _____________________________________________________________________________
+string LoadURL::getCacheKeyImpl() const {
+  // TODO<qup42> do caching based on ETag, Last-Modified or similar
+  if (RuntimeParameters().get<"cache-load-results">()) {
+    return absl::StrCat("LOAD URL ", loadURLClause_.url_.asString(),
+                        loadURLClause_.silent_ ? " SILENT" : "");
+  }
+  return absl::StrCat("LOAD URL ", cacheBreaker_);
+}
+
+// _____________________________________________________________________________
+string LoadURL::getDescriptor() const {
+  return absl::StrCat("LOAD URL ", loadURLClause_.url_.asString());
+}
+
+// _____________________________________________________________________________
+size_t LoadURL::getResultWidth() const { return 3; }
+
+// _____________________________________________________________________________
+size_t LoadURL::getCostEstimate() {
+  // TODO: For now, we don't have any information about the cost at query
+  // planning time, so we just return ten times the estimated size.
+  return 10 * getSizeEstimateBeforeLimit();
+}
+
+// _____________________________________________________________________________
+uint64_t LoadURL::getSizeEstimateBeforeLimit() {
+  // TODO: For now, we don't have any information about the result size at
+  // query planning time, so we just return `100'000`.
+  return 100'000;
+}
+
+// _____________________________________________________________________________
+float LoadURL::getMultiplicity(size_t) {
+  // TODO: For now, we don't have any information about the multiplicities at
+  // query planning time, so we just return `1` for each column.
+  return 1;
+}
+
+// _____________________________________________________________________________
+bool LoadURL::knownEmptyResult() { return false; }
+
+// _____________________________________________________________________________
+std::unique_ptr<Operation> LoadURL::cloneImpl() const {
+  auto load = std::make_unique<LoadURL>(_executionContext, loadURLClause_);
+  load->cacheBreaker_ = cacheBreaker_;
```

Comment:
That is wrong, clone should be a copletely decoupled deep clone, the cache breaker should definitely be different
(even though it is unlikely, that we would ever encounter that bug....

Reply:
`Operation::clone` checks that `getCacheKey` of a clone is the same as the original. This invariant would then no longer hold. (The same changes also go for the `Service` operation which does the same; I'll update that once we have decided what to do).

Reply:
I see, but I have just thought of a possible solution:
1. Override the virtual `canResultBeCached` method to always return `false`. 
2. Adapt the check in `Operation::clone` to skip the cache key check, if we can't cache the new value anyway.
3. Make sure to add a unit test for the `clone` (I am not sure if @RobinTF has put them in one place, or with the individual operations, or whether you have already done them).

Reply:
Implemented it that way for now, but I have some follow up questions. Afaik this disables all caching of this operations, so the cache breaker and `RuntimeParameter` in `getCacheKey` wouldn't be needed. Returning the state of the runtime parameter seems to make more sense to me. There also shouldn't be any problems when switching because the cache keys between caching enabled/disabled are disjunct.

Reply:
Yes, 
What about
1. At construction time you cache the value of the runtime parameter.
2. your "canResultBeCached" override returns exactly that value

This gives a fairly reasonable behavior wrt. concurrent changes of the parameter etc.

Reply:
Another strange thing `disableStoringInCache` is `public`, `virtual` and `final` meaning it cannot be overwritten. It changes the value of `Operation::canResultBeCached_` which is a private member. So for all non default behavior except `return false` of `canResultBeCached` (as we have here) this breaks that function. What was the reason for making it final?